internal/handler/handler.go

```
@@ -65,31 +85,230 @@ func New(opts *NewOpts) (*Handler, error) {
 		b = oplog.NewBackend(b, opts.L.Named("oplog"))
 	}
 
+	if opts.CappedCleanupPercentage > 100 {
```

I think we should return an error there (and also check for `> 99`, not 100)